src/main/java/spoon/FluentLauncher.java

```diff
@@ -0,0 +1,160 @@
+package spoon;
+
+import java.io.File;
+import java.util.List;
+
+import spoon.compiler.Environment;
+import spoon.compiler.SpoonResource;
+import spoon.processing.Processor;
+import spoon.reflect.CtModel;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.visitor.Filter;
+import spoon.reflect.visitor.PrettyPrinter;
+import spoon.support.JavaOutputProcessor;
+
+/**
+ * TODO: doc
+ */
+public class FluentLauncher {
+
+	Launcher launcher;
+
+	public FluentLauncher() {
+		this.launcher = new Launcher();
+	}
+
+	/**
+	 * @param args
+	 * @see spoon.Launcher#run(java.lang.String[])
+	 */
+
+	public void run(String[] args) {
+		launcher.run(args);
+	}
+
+	public void printUsage() {
+		launcher.printUsage();
+	}
+
+	public FluentLauncher inputResource(String path) {
+		launcher.addInputResource(path);
+		return this;
+	}
+
+	public FluentLauncher inputResource(SpoonResource resource) {
+		launcher.addInputResource(resource);
+		return this;
+	}
+
+	public FluentLauncher processor(String name) {
+		launcher.addProcessor(name);
+		return this;
+	}
+
+	public <T extends CtElement> FluentLauncher processor(Processor<T> processor) {
+		launcher.addProcessor(processor);
+		return this;
+	}
+
+	public FluentLauncher templateResource(SpoonResource resource) {
+		launcher.addTemplateResource(resource);
+		return this;
+	}
+
+	// TODO: das sollte in Methoden zum setzen umgewandelt werden
+	public Environment getEnvironment() {
+		return launcher.getEnvironment();
+	}
+
+	public String toString() {
+		return launcher.toString();
+	}
+
+	// needed???
+	public SpoonModelBuilder compiler(Factory factory) {
```

Comment:
How does a `SpoonModelBuilder` differ from the `FluentLauncher`? Conceptually, to me, they seem like they are trying to accomplish the same task: using a builder pattern to configure Spoon. Exposing both may lead to some confusion.

Reply:
The SpoonModelBuilder interace doesn't expose a fluent api.